ui/app/components/secret-edit.js

```diff
@@ -294,16 +294,22 @@ export default Component.extend(FocusOnInsertMixin, WithNavToNearestAncestor, {
     },
 
     handleCopyError() {
-      this.flashMessages.error('Could Not Copy Wrapped Data');
+      this.flashMessages.danger('Could Not Copy Wrapped Data');
       this.send('clearWrappedData');
     },
 
     createOrUpdateKey(type, event) {
       event.preventDefault();
       let model = this.modelForData;
+      let secret = this.model;
       // prevent from submitting if there's no key
-      // maybe do something fancier later
       if (type === 'create' && isBlank(model.path || model.id)) {
+        this.flashMessages.danger('Please provide a path for the secret');
+        return;
+      }
+      const maxVersions = secret.get('maxVersions');
+      if (maxVersions > 9999999999999999) {
```

Comment:
you might want to drop a comment in here to explain the magic number, but otherwise this seems like a reasonable solution to me 👍